core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/UserConfigImpl.java

```diff
@@ -0,0 +1,95 @@
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component(service = UserConfig.class)
+@Slf4j
+public class UserConfigImpl implements UserConfig {
+
+    @Reference
+    private RepositoryHelper repositoryHelper;
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends Annotation> T apply(T baseConfig) {
+        Handler<T> handler = new Handler<>(baseConfig, getProperties(baseConfig.annotationType().getName()));
+        Object proxy = Proxy.newProxyInstance(
+                baseConfig.annotationType().getClassLoader(),
+                new Class[]{baseConfig.annotationType()},
+                handler);
+        return (T) proxy;
+    }
+
+    private Map<String, Object> getProperties(String configId) {
+        try(ResourceResolver resourceResolver = repositoryHelper.getServiceResourceResolver()){
+            Resource resource = resourceResolver.getResource("/conf/etoolbox-link-inspector/data/optionsconfig/" + StringUtils.substringBeforeLast(configId, "$"));
+            if (resource == null) {
+                return Collections.emptyMap();
+            }
+            Map<String, Object> properties = new HashMap<>();
+            properties.putAll(resource.adaptTo(ValueMap.class));
+            return properties;
+        }
+    }
+
+    @RequiredArgsConstructor
+    private static class Handler<T extends Annotation> implements InvocationHandler {
+        private static final String METHOD_TO_STRING = "toString";
+        private static final String METHOD_ANNOTATION_TYPE = "annotationType";
+
+        private final T source;
+        private final Map<String, Object> overriddenProperties;
+
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args) {
+            if (METHOD_TO_STRING.equals(method.getName())) {
+                return source.toString();
+            }
+            if (METHOD_ANNOTATION_TYPE.equals(method.getName())) {
+                return source.annotationType();
+            }
+            Object override = MapUtils.getObject(overriddenProperties, method.getName());
+            if (isNotBlank(override)) {
+                if (NumberUtils.isCreatable(override.toString())) {
+                    return Integer.parseInt(override.toString());
+                } else if ("false".equals(override) || "true".equals(override)) {
```

Comment:
👍 Good one! But `StringUtils.equalsAnyIgnoreCase(override.toString(), "true", "false")`.
That's because "override" is an Object. It could be a wrapper over a truly Boolean value. And if "override" is truly stored as a Boolean value (e.g. `true`), `"true".equals(override) == false` because this is not Javascript ))

Reply:
done